src/t_stream.c

```diff
@@ -2880,17 +2903,29 @@ void xpendingCommand(client *c) {
     long long count = 0;
     long long minidle = 0;
     int startex = 0, endex = 0;
+    int mkgroup = 0;   
 
+    robj *o = lookupKeyRead(c->db,key);
+    streamCG *group;
+
+    if (checkType(c,o,OBJ_STREAM)) return;
+    if (o == NULL) {
+        addReplyErrorFormat(c, "No such key '%s'",
+                                     (char*)key->ptr);
+        return;
+    }
```

Comment:
i'd rather move the lookup (KSN) and type check to after processing the arguments and syntax errors.